### PR TITLE
[INLONG-624]Go SDK consumer start interface

### PR DESCRIPTION
**Pull request by TszKitLo40**
Signed-off-by: Zijie Lu <wslzj40@gmail.com>

### CODE REVIEW

**Review by gosonzhang — APPROVED** [review]
+1

File: tubemq-client-twins/tubemq-client-go/client/consumer.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Package client defines the api and information
+// which can be exposed to user.
+package client
+
+const (
+	tubeMQClientVersion = "0.1.0"
```
Comment (charlely) [inline tubemq-client-twins/tubemq-client-go/client/consumer.go:23]: Need a separate file to define version

File: tubemq-client-twins/tubemq-client-go/client/consumer_impl.go [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+func (c *consumer) register2Master(needChange bool) error {
+	if needChange {
+		node, err := c.selector.Select(c.config.Consumer.Masters)
+		if err != nil {
+			return err
+		}
+		c.master = node
+	}
+	for c.master.HasNext {
+		ctx, cancel := context.WithTimeout(context.Background(), c.config.Net.ReadTimeout)
```
Comment (charlely) [inline tubemq-client-twins/tubemq-client-go/client/consumer_impl.go:121]: Need use defer cancel(), can define other function to register one master.

File: tubemq-client-twins/tubemq-client-go/client/consumer_impl.go [hunk 3]
```diff
@@ -0,0 +1,356 @@
...
+
+func (c *consumer) processRebalanceEvent() {
+	for {
+		event := c.rmtDataCache.TakeEvent()
+		if event.GetEventStatus() == int32(util.InvalidValue) && event.GetRebalanceID() == util.InvalidValue {
+			break
+		}
+		c.rmtDataCache.ClearEvent()
+		switch event.GetEventType() {
+		case 2, 20:
```
Comment (charlely) [inline tubemq-client-twins/tubemq-client-go/client/consumer_impl.go:211]: const value cann't use magic number.

File: tubemq-client-twins/tubemq-client-go/client/consumer_impl.go [hunk 4]
```diff
@@ -0,0 +1,356 @@
...
+	panic("implement me")
+}
+
+// GetCurrConsumedInfo implementation of TubeMQ consumer.
+func (c *consumer) GetCurrConsumedInfo() (map[string]*ConsumerOffset, error) {
+	panic("implement me")
+}
+
+func (c *consumer) processRebalanceEvent() {
+	for {
```
Comment (charlely) [inline tubemq-client-twins/tubemq-client-go/client/consumer_impl.go:204]: Why not use channels

Comment (charlely) [reply tubemq-client-twins/tubemq-client-go/client/consumer_impl.go:204]: The for loop will run to full cpu

File: tubemq-client-twins/tubemq-client-go/client/consumer_impl.go [hunk 5]
```diff
@@ -0,0 +1,356 @@
...
+		event := c.rmtDataCache.TakeEvent()
+		if event.GetEventStatus() == int32(util.InvalidValue) && event.GetRebalanceID() == util.InvalidValue {
+			break
+		}
+		c.rmtDataCache.ClearEvent()
+		switch event.GetEventType() {
+		case 2, 20:
+			c.disconnect2Broker(event)
+			c.rmtDataCache.OfferEventResult(event)
+		case 1, 10:
```
Comment (charlely) [inline tubemq-client-twins/tubemq-client-go/client/consumer_impl.go:214]: const value cann't use magic number.

File: tubemq-client-twins/tubemq-client-go/client/consumer.go [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+// ConsumerResult of a consumption.
+type ConsumerResult struct {
+}
+
+// ConsumerOffset of a consumption,
+type ConsumerOffset struct {
+}
+
+var clientIndex uint64
```
Comment (charlely) [inline tubemq-client-twins/tubemq-client-go/client/consumer.go:34]: This should be the allocator of clientid, index is more like array coordinates.

**Review by charlely — APPROVED** [review]

**Review by charlely — APPROVED** [review]
LGTM

**Pull request merged**